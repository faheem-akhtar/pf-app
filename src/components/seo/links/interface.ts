import { SeoLinksLinkInterface } from './link.interface';

export interface SeoLinksInterface {
  /**
   * Popular searches
   */
  popularSearches?: SeoLinksLinkInterface;

  /**
   * Nearby areas
   */
  nearbyAreas?: SeoLinksLinkInterface;

  /**
   * Alternate category
   */
  alternateCategory?: SeoLinksLinkInterface;
}
