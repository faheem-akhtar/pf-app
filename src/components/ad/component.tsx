import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
