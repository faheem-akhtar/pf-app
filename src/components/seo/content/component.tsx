import { FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import { SeoContentComponentPropsInterface } from './component-props.interface';
import styles from './seo-content.module.scss';

export const SeoContentComponent: FunctionComponent<SeoContentComponentPropsInterface> = ({
  heading,
  content,
  image,
}) => (
  <div className={styles.container}>
    {heading && <h2 className={styles.heading}>{heading}</h2>}
    {content && (
      <div className={styles.body}>
        {image && (
          <picture
            className={domClassMerge(styles['body-picture'], {
              [styles['body-picture--right']]: !image.align || image.align === 'right',
              [styles['body-picture--left']]: image.align === 'left',
            })}
          >
            <img className={styles['body-picture-img']} src={image.url} alt={image.alt} />
          </picture>
        )}
        <div className={styles['body-html']} dangerouslySetInnerHTML={{ __html: content }} />
      </div>
    )}
  </div>
);
