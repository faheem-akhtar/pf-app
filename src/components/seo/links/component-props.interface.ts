import { SeoLinksInterface } from './interface';

export interface SeoLinksComponentPropsInterface extends SeoLinksInterface {}
