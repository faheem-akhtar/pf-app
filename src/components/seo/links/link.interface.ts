import { SeoLinkInterface } from 'components/seo/link.interface';

export interface SeoLinksLinkInterface {
  /**
   * Title of the category
   * @example Popular searches
   */
  title: string;

  /**
   * List of seo links
   */
  links: SeoLinkInterface[];
}
