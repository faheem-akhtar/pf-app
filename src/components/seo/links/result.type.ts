import { SeoLinksAttributeInterface } from './attribute.interface';

export type SeoLinksResultType = {
  /**
   * Popular searches
   */
  popularSearches?: SeoLinksAttributeInterface;

  /**
   * Nearby areas
   */
  nearbyAreas?: SeoLinksAttributeInterface;

  /**
   * Alternate category
   */
  alternateCategory?: SeoLinksAttributeInterface;
};
