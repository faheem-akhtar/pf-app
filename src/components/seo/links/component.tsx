import { FunctionComponent } from 'react';

import { AccordionComponent } from 'library/accordion/component';
import { AccordionItemComponent } from 'library/accordion/item/component';

import styles from '../seo.module.scss';
import { SeoLinksComponentPropsInterface } from './component-props.interface';

export const SeoLinksComponent: FunctionComponent<SeoLinksComponentPropsInterface> = ({
  popularSearches,
  nearbyAreas,
  alternateCategory,
}) => (
  <AccordionComponent>
    {popularSearches && (
      <AccordionItemComponent title={popularSearches.title} expanded>
        {popularSearches.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
        {alternateCategory?.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
      </AccordionItemComponent>
    )}

    {nearbyAreas && (
      <AccordionItemComponent title={nearbyAreas.title}>
        {nearbyAreas.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
      </AccordionItemComponent>
    )}
  </AccordionComponent>
);
