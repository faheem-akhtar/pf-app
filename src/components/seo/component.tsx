import { FunctionComponent } from 'react';

import { SeoComponentPropsInterface } from './component-props.interface';
import { SeoContentComponent } from './content/component';
import { SeoLinksComponent } from './links/component';
import styles from './seo.module.scss';

export const SeoComponent: FunctionComponent<SeoComponentPropsInterface> = ({
  popularSearches,
  nearbyAreas,
  alternateCategory,
  content,
}) => (
  <section id='cms-content'>
    {(content?.primaryContent || content?.secondaryContent) && <hr className={styles.hr} />}
    {content?.primaryContent && (
      <SeoContentComponent
        heading={content.primaryHeading}
        content={content.primaryContent}
        image={
          content.primaryImageUrl
            ? {
                url: content.primaryImageUrl,
                alt: content.primaryImageAlt,
              }
            : undefined
        }
      />
    )}
    {content?.secondaryContent && (
      <SeoContentComponent
        heading={content.secondaryHeading}
        content={content.secondaryContent}
        image={
          content.secondaryImageUrl
            ? {
                url: content.secondaryImageUrl,
                alt: content.secondaryImageAlt,
                align: 'left',
              }
            : undefined
        }
      />
    )}
    <SeoLinksComponent
      popularSearches={popularSearches}
      nearbyAreas={nearbyAreas}
      alternateCategory={alternateCategory}
    />
  </section>
);
