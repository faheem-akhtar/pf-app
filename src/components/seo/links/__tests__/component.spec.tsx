import { render } from '@testing-library/react';

import { seoLinksStub } from 'stubs/seo/links.stub';

import { SeoLinksComponent } from '../component';

describe('SeoLinksComponent', () => {
  it('should not render anything when both links are empty', () => {
    const { container } = render(<SeoLinksComponent />);

    expect(container).toBeEmptyDOMElement();
  });

  it('renders without throwing any errors', () => {
    const { container } = render(<SeoLinksComponent {...seoLinksStub()} />);

    expect(container).toMatchSnapshot();
  });
});
