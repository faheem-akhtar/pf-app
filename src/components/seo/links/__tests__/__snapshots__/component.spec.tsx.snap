// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoLinksComponent renders without throwing any errors 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <p
        class="title"
      >
        Popular searches
      </p>
      <svg
        class="icon icon--expanded"
        viewBox="4.5 8 16 8"
      >
        <path
          clip-rule="evenodd"
          d="M4.64645 8.14645C4.47288 8.32001 4.4536 8.58944 4.58859 8.78431L4.64645 8.85355L11.6464 15.8536C11.82 16.0271 12.0894 16.0464 12.2843 15.9114L12.3536 15.8536L19.3536 8.85355C19.5488 8.65829 19.5488 8.34171 19.3536 8.14645C19.18 7.97288 18.9106 7.9536 18.7157 8.08859L18.6464 8.14645L12 14.793L5.35355 8.14645C5.17999 7.97288 4.91056 7.9536 4.71569 8.08859L4.64645 8.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </header>
    <div
      class="panel"
    >
      <a
        class="link"
        href="/en/buy/apartments-for-sale.html"
      >
        Apartments for sale
      </a>
      <a
        class="link"
        href="/en/buy/dubai/properties-for-sale.html"
      >
        Properties for sale in Dubai
      </a>
      <a
        class="link"
        href="/en/buy/properties-for-sale.html"
      >
        Properties for sale
      </a>
    </div>
  </div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <p
        class="title"
      >
        Nearby Areas
      </p>
      <svg
        class="icon"
        viewBox="4.5 8 16 8"
      >
        <path
          clip-rule="evenodd"
          d="M4.64645 8.14645C4.47288 8.32001 4.4536 8.58944 4.58859 8.78431L4.64645 8.85355L11.6464 15.8536C11.82 16.0271 12.0894 16.0464 12.2843 15.9114L12.3536 15.8536L19.3536 8.85355C19.5488 8.65829 19.5488 8.34171 19.3536 8.14645C19.18 7.97288 18.9106 7.9536 18.7157 8.08859L18.6464 8.14645L12 14.793L5.35355 8.14645C5.17999 7.97288 4.91056 7.9536 4.71569 8.08859L4.64645 8.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </header>
  </div>
</div>
`;
